design-proposals/klip-38-variable-substitution.md

```diff
@@ -264,7 +264,21 @@ Scope: SESSION
 In Java API client, the session is opened when `Client.create()` is called, and closed when `Client.close()` is called. Variables
 may be defined using a client method instead of a new syntax. This is to ensure users know the scope starts after the `create()` call.
 
-TBD: Define a Java API method for variable substitution.
+Users will define variables using the functions, `define(variable_name, value)` and `undefine(variable_name)`.
+The client will keep track of the calls made in a map. Alternatively, the `executeStatement` and `executeQuery`
+functions could accept a third parameter, `sessionVariables` that is a map of variables to variable definitions.
+
+There are two options for executing the variable execution. The first is to do it in the client by
+using the variable substitutor in `ksqldb-parser`. The disadvantages to this are that `ksqldb-parser`
+would be a dependancy of the Java client, and that the client and server versions would have
+to match in order for the parser to work.
+
+The other option is to update the endpoints, `/ksql` and `/query-streams` to accept `sessionVariables`
```

Comment:
As mentioned above, there are other endpoints that we'd want to support variable substitution on as well. Definitely `/inserts-stream` for now, potentially others in the future.

Reply:
Added `/inserts-stream` to the klip. The other two endpoints used by the client, `/close-query` and `/info` will be left as is (although technically the user could pass in a variable to `/close-query`, I'm not sure how useful supporting this actually is.)